src/test/java/com/pmarko09/medical_clinic_proxy/api/DoctorApiTest.java

```diff
@@ -0,0 +1,104 @@
+package com.pmarko09.medical_clinic_proxy.api;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.pmarko09.medical_clinic_proxy.model.dto.VisitDto;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.cloud.contract.wiremock.AutoConfigureWireMock;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@AutoConfigureWireMock(port = 8089)
+public class DoctorApiTest {
+
+    @Autowired
+    WireMockServer wireMockServer;
+
+    @Autowired
+    RestTemplate restTemplate;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @BeforeEach
+    public void setUp() {
+        wireMockServer.start();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        wireMockServer.resetAll();
+        wireMockServer.stop();
+    }
+
+    @Test
+    void getAvailableVisitsForDoctor_DataCorrect_VisitDtosReturned() throws JsonProcessingException {
+        VisitDto visitDto1 = new VisitDto(1L,
+                LocalDateTime.of(2024, 9, 22, 16, 30, 00),
+                LocalDateTime.of(2024, 9, 22, 17, 00, 00),
+                3L, 7L);
+        VisitDto visitDto2 = new VisitDto(2L,
+                LocalDateTime.of(2024, 10, 22, 16, 30, 00),
+                LocalDateTime.of(2024, 10, 22, 17, 00, 00),
+                3L, 7L);
+        List<VisitDto> visitDtos = new ArrayList<>(List.of(visitDto1, visitDto2));
+
+        wireMockServer.stubFor(WireMock.get("/appointments/doctor/3/available")
+                .willReturn(WireMock.aResponse()
+                        .withStatus(HttpStatus.OK.value())
+                        .withHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                        .withBody(objectMapper.writeValueAsString(visitDtos))));
```

Comment:
tutaj, zamiast tworzyć liste wyżej, możesz po prostu napisać

.withBody(objectMapper.writeValueAsString(List.of(visitDto1, visitDto2)))));

Reply:
racja, ale moze lekko przejrzysciej jest tak jak ja mam?